tools/mo/openvino/tools/mo/front/extractor.py

```diff
@@ -924,21 +930,34 @@ def add_input_op(graph: Graph, node_id: str, port: int = 0, data: bool = False,
     input_op = Parameter(graph, dict(shape=shape, user_shape=user_shape, data_type=data_type, initial_node_name=node_id,
                                      name=get_new_placeholder_name(node_id, is_out_port, port)))
 
-    fw_name = Node(graph, node_id).soft_get('name')
+    tensor_name = Node(graph, node_id).soft_get('name') + ":" + str(port)
+    fw_info = [(Node(graph, node_id).soft_get('name'), tensor_name)]
+    if is_out_port and port == 0:
+        tensor_name_no_port = Node(graph, node_id).soft_get('name')
+        # TODO: This can be optimized. Tensor names can be stored as set, which is initialized after model loading.
+        graph_tensor_names = graph.get_tensor_names_set()
+        if tensor_name_no_port in graph_tensor_names:
+            log.warning('Could not add user defined input name {} to tensor names list of as '
+                        'graph contains tensor name with same name.'.format(tensor_name_no_port))
```

Comment:
nit: propose to create a routine in Graph class `has_tensor_name` that will check if such tensor name exists in a graph. It will be instead of `get_tensor_names_set`

Reply:
Done.